crates/router/src/configs/settings.rs

```diff
@@ -785,6 +787,19 @@ pub struct PayPalOnboarding {
     pub enabled: bool,
 }
 
+#[cfg(feature = "tls")]
+#[derive(Debug, Deserialize, Clone)]
+pub struct ServerTls {
+    /// Port to host the TLS secure server on
+    pub port: u16,
+    /// Use a different host (optional) (defaults to the host provided in [`Server`] config)
+    pub host: Option<String>,
+    /// private key file path associated with TLS (path to the private key file (`pem` format))
+    pub private_key: String,
+    /// certificate file associated with TLS (path to the certificate file (`pem` format))
+    pub certificate: String,
```

Comment:
nit: can we instead have `PathBuf` here? not a critical change, but good to have types analogous to the data that they represent